src/solvers/smt2/smt2_conv.cpp

```diff
@@ -2214,6 +2214,10 @@ void smt2_convt::convert_typecast(const typecast_exprt &expr)
         convert_typecast(tmp);
       }
     }
+    else if(src_type.id() == ID_empty)
+    {
+      convert_expr(src);
+    }
```

Comment:
This might actually fix some of the `broken-smt-backend` tests - would you mind testing that?